### ISSUE TITLE
Code intelligence on lib projects

### ISSUE THREAD

**Issue by Harbs** [BowlerHatLLC/vscode-nextgenas#265]
I'm not sure what changed, but for some reason, the `asconfig.json` files in the Royale projects stopped working.

Code intelligence no longer works. For example, I get the following in the Problems pane:
`DropDownList.as is not located in the project's source path. Code intelligence will not be available for this file.`

The config file is using `include-classes` and `include-sources`. This used to work.

Adding the following allows code intelligence, but causes hundreds of flash warnings:
````
        "source-path": [
            "src/main/royale"
        ],
````
I can't figure out how to configure the asconfig file to make it work again.

**Comment by Harbs** [comment]
Here's one of the files:
https://github.com/apache/royale-asjs/blob/develop/frameworks/projects/Basic/asconfig.json

**Comment by Harbs** [comment]
I'm getting the errors for classes which are SWF-only. (i.e. `org.apache.royale.html.beads.BackgroundImageBead`)

Interestingly, making SWF the first target does not help for that. I don't know why.

It used to work because the classes were brought in by BasicClasses which only referenced the SWF classes in COMPILE:SWF blocks.

**Comment by Harbs** [comment]
Here's a couple of example messages:
`Definition flash.display.Sprite could not be found.`
`Type was not found or was not a compile-time constant: Sprite.`

There are many like these.

**Comment by joshtynjala** [comment]
> Interestingly, making SWF the first target does not help for that. I don't know why.

That's a bug in v0.13. It's already fixed and ready to go into the upcoming v0.14. Carlos reported this one the other day.

> I'm getting the errors for classes which are SWF-only. (i.e. org.apache.royale.html.beads.BackgroundImageBead)

You don't seem to be using conditional compilation to exclude this class, so VSCode is behaving correctly.

Using the `COMPILE::SWF` conditional compilation constant seems to do the trick:

``` actionscript
package org.apache.royale.html.beads
{
	COMPILE::SWF
	{
		import flash.display.Bitmap;
		import flash.display.Loader;
		import flash.display.LoaderInfo;
		import flash.display.Sprite;
		import flash.events.Event;
		import flash.events.IOErrorEvent;
		import flash.net.URLRequest;
	}
	
	import org.apache.royale.core.IBead;
	import org.apache.royale.core.IStrand;
	import org.apache.royale.core.UIBase;
	import org.apache.royale.core.ValuesManager;

	import org.apache.royale.html.beads.IBackgroundBead;
	
	COMPILE::SWF
	/**
	 *  The BackgroundImageBead is used to render an image as the background to any component
	 *  that supports it, such as Container.
	 * 
	 *  Note that this bead is for ActionScript only since CSS/HTML allows this just by specifying
	 *  a background image in the style selector. To use this bead, place a ClassReference to it
	 *  within @media -royale-swf { } group in the CSS declarations.
	 *  
	 *  @langversion 3.0
	 *  @playerversion Flash 10.2
	 *  @playerversion AIR 2.6
	 *  @productversion Royale 0.0
	 */
	public class BackgroundImageBead implements IBead, IBackgroundBead
	{
//...
```

What's strange to me is that you're saying that these errors are new, but I always remember seeing these sorts of errors whenever I tried to open Royale framework projects.

One thing worth noting is that VSCode will show errors for all files in the project's source path, even if they are not referenced anywhere (in other words, the compiler wouldn't include them in its generated SWF or JS). This behavior is a bit different from Flash Builder, where I think errors are displayed only for files that are referenced (andunreferenced files will show errors only if open in an editor).

**Comment by joshtynjala** [comment]
I'll take a look and see if I can figure out how to make `include-classes` work without `source-path`.
